jscomp/bsb/bsb_ninja_file_groups.ml

```diff
@@ -108,6 +108,10 @@ let emit_module_build
       | Res, Reason -> { impl = res_suffixes.impl; intf = re_suffixes.intf }
       | Ml, Ml | Reason, Reason | Res, Res -> assert false
   in
+  let error_syntax_kind =
+    match module_info.syntax_kind with
+    | Same syntax_kind -> syntax_kind
+    | Different { impl } -> impl in
```

Comment:
I always prefer the implementation errors here, but if we compile them in two different rules it should be possible to emit different errors, but that would be more work.